### DISCUSSION TITLE
How to use the command after installing the code without the "key file"?

### DISCUSSION THREAD

**Discussion by VarunGit19** [ElDavoo/wa-crypt-tools#151] [General]
I found out about this repo just now. I have an crypt15 file and I don't have the key file but I have the actual key (which is the password). In the command given in the instructions, it requires the key file. I stored the password in a txt file and gave it to the code but it shows this error: 
keyfactory.py:43 	: [C] The keyfile is not a valid Java object: Invalid file magic: 0x3330
keyfactory.py:55 	: [C] Unrecognized key file format.

so what is the format of the key file? Can't I directly give it to the code and run it using a command? 

Actually, I have a backup of my data when I had an E2E backup enabled. After a while, I disabled it and when I reinstalled whatsapp, it's not even checking for the local backups. So, since I have the backup file and the password is known to me, I want to extract the data

**Comment by ElDavoo** [comment]
Whoops. It was written in the instructions, but i deleted it.
Instead of a file name, you can paste the key directly.
Or, you can use the wacreatekey utility.

**Reply by VarunGit19** [reply]
So,
$ wadecrypt encrypted_backup.key msgstore.db.crypt15 msgstore.db
key15.py:51     : [I] Crypt15 / Raw key loaded
wadecrypt.py:271        : [I] Done

in this code, instead of the '.key' file, can I directly give my key in string format?

**Reply by ElDavoo** [reply]
yes

**Reply by VarunGit19** [reply]
keyfactory.py:23 	: [C] The key file specified does not exist.
    If you tried to specify the key directly, note it should be 64 characters long and not 7 characters long.
    

 this is the error I am getting. I don't have the 64 bit key

**Reply by ElDavoo** [reply]
oh no, the password is not the key file. https://github.com/ElDavoo/wa-crypt-tools?tab=readme-ov-file#can-i-use-the-password-to-decrypt-the-database

**Reply by ElDavoo** [reply]
see also https://github.com/ElDavoo/wa-crypt-tools?tab=readme-ov-file#where-do-i-get-the-keyfile

**Reply by VarunGit19** [reply]
No no, the command I used was:
!wadecrypt "my password" msgstore-2021-10-01.db.crypt15 msgstore.db

**Reply by ElDavoo** [reply]
please read the above links,

**Reply by VarunGit19** [reply]
Ohh okay.. that means nothing can be done now.. I have uninstalled and reinstalled many times after that and changed the password as well as disabled it. Also I don't have a rooted Android device
